Yosemite/YosemiteTests/Stores/ShippingLabelStoreTests.swift

```diff
@@ -838,25 +838,60 @@ final class ShippingLabelStoreTests: XCTestCase {
                                     labelIDs: [13579],
                                     thenReturn: .success([labelStatusResponse]))
         let store = ShippingLabelStore(dispatcher: dispatcher, storageManager: storageManager, network: network, remote: remote)
-
-        // When
-        var purchaseResult: Result<[Yosemite.ShippingLabel], Error>? = waitFor(timeout: 6.0) { promise in
-            DispatchQueue.main.asyncAfter(deadline: .now() + 5.9) {
-                promise(nil)
-            }
-        }
+        var purchaseResult: Result<[Yosemite.ShippingLabel], Error>? = nil
```

Comment:
I think what the version before did was accounting for the internal delays and forcing retries to check the status by waiting for 6 seconds (2s for the first request + 1s after each status request run a few times, plus buffer)

It worked, but I find this clearer:

```swift
var purchaseResult: Result<[Yosemite.ShippingLabel], Error>? = nil
// ...
let exp = expectation(
    for: NSPredicate(block: { _,_ in
      switch purchaseResult {
        case .failure: return true
        default: return false
      }
    }),
    evaluatedWith: nil
)
exp.isInverted = true
// ...
wait(for: [exp], timeout: 6.0)
```

If we'll find ourselves using the expectation with predicate strategy a few more times, we could add some syntax sugar to make it more compact.